File: rxhttp/src/main/java/rxhttp/wrapper/parse/SimpleParser.java
package rxhttp.wrapper.parse;


import java.io.IOException;
import java.lang.reflect.Type;

import okhttp3.Response;

/**
 * 将Response对象解析成泛型T对象
 * User: ljx
 * Date: 2018/10/23
 * Time: 13:49
 */
public class SimpleParser<T> extends AbstractParser<T> {

    protected SimpleParser() {
        super();
    }

    private SimpleParser(Type type) {
        super(type);
    }

    public static <T> SimpleParser<T> get(Class<T> type) {
        return new SimpleParser<>(type);
    }


    @Override
    public T onParse(Response response) throws IOException {
        return convert(response, mType);
    }
}

File: rxhttp/src/main/java/rxhttp/wrapper/parse/SimpleParser.kt
package rxhttp.wrapper.parse

import okhttp3.Response
import rxhttp.wrapper.utils.convert
import java.io.IOException
import java.lang.reflect.Type

/**
 * Convert [Response] to [T]
 * User: ljx
 * Date: 2018/10/23
 * Time: 13:49
 */
open class SimpleParser<T> : TypeParser<T> {
    protected constructor() : super()
    constructor(type: Type) : super(type)

    @Throws(IOException::class)
    override fun onParse(response: Response): T = response.convert(types[0])
}